### PR TITLE
Jacobian tests

### PR DESCRIPTION
**Pull request by LeoMaglanoc**
new tests to compare jacobian from sara-shield to symbolic jacobian from 3 joint planar robot (Siciliano - 2009 - Robotics modelling, planning and control, p. 69 & 114)

### CODE REVIEW

File: safety_shield/include/safety_shield/robot_reach.h [hunk 1]
```diff
@@ -260,6 +260,10 @@ class RobotReach {
     return cross;
   }
 
+  inline reach_lib::Point getTcpOfVelocityCapsule() {
```
Comment (JakobThumm) [inline safety_shield/include/safety_shield/robot_reach.h]: I would not introduce this function.

Comment (JakobThumm) [reply safety_shield/include/safety_shield/robot_reach.h]: You could introduce the function `getRobotVelocityCapsules()`. That would make more sense.

File: safety_shield/tests/src/robot_reach_tests.cc [hunk 2]
```diff
@@ -274,6 +275,78 @@ TEST_F(RobotReachTest, ApproximateVelOfCapsuleTest1) {
...
+    return jacobian;
+}
+
+TEST_F(RobotReachTestVelocity, JacobianSicilianoTest0) {
+    std::vector<double> q = {0.0, 0.0, 0.0};
+    robot_reach_->calculateAllTransformationMatricesAndCapsules(q);
+    reach_lib::Point point = robot_reach_->getTcpOfVelocityCapsule();
+    Eigen::Matrix<double, 6, Eigen::Dynamic> jacobian = robot_reach_->getJacobian(2, point);
+    Eigen::Matrix<double, 6, Eigen::Dynamic> expect = jacobian_tcp_siciliano(q);
+    ASSERT_TRUE((jacobian - expect).norm() < 1e-8);
```
Comment (JakobThumm) [inline safety_shield/tests/src/robot_reach_tests.cc]: If something fails here, you get no feedback, why. 
I would use 3*6 `EXPEXT_NEAR()` statements. 
This looks a bit ridiculous but gives you the most insights why the test failed.
You could maybe write a function for these 18 EXPECTS to reuse them later.

File: safety_shield/tests/src/robot_reach_tests.cc [hunk 3]
```diff
@@ -274,6 +275,78 @@ TEST_F(RobotReachTest, ApproximateVelOfCapsuleTest1) {
...
+    std::vector<double> q = {1.247, 0.373, 2.142};
+    robot_reach_->calculateAllTransformationMatricesAndCapsules(q);
+    reach_lib::Point point = robot_reach_->getTcpOfVelocityCapsule();
+    Eigen::Matrix<double, 6, Eigen::Dynamic> jacobian = robot_reach_->getJacobian(2, point);
+    Eigen::Matrix<double, 6, Eigen::Dynamic> expect = jacobian_tcp_siciliano(q);
+    ASSERT_TRUE((jacobian - expect).norm() < 1e-8);
+}
+
+TEST_F(RobotReachTestVelocity, JacobianSicilianoTest4) {
+    std::vector<double> q = {10.216, 4.263, 1.984};
```
Comment (JakobThumm) [inline safety_shield/tests/src/robot_reach_tests.cc]: Please also test negative values.

**Comment by LeoMaglanoc** [comment]
- introduced `getRobotVelocityCapsules` 
- testing negative values 
- testing each jacobian entry individually
- applied formatter to `safety_shield.h`

File: safety_shield/include/safety_shield/robot_reach.h [hunk 4]
```diff
@@ -260,6 +260,13 @@ class RobotReach {
     return cross;
   }
 
+  /**
+   * @brief returns robot capsules for velocity
+   * @return robot capsules for velocity
+   */
+  inline std::vector<occupancy_containers::capsule::Capsule>& getRobotCapsulesForVelocity() {
```
Comment (JakobThumm) [inline safety_shield/include/safety_shield/robot_reach.h]: I would not return a reference. Either return a pointer to the vector or the vector itself.
So 
`inline std::vector<occupancy_containers::capsule::Capsule> getRobotCapsulesForVelocity() {`
or 
 `inline std::vector<occupancy_containers::capsule::Capsule>* getRobotCapsulesForVelocity() {`

File: safety_shield/src/robot_reach.cc [hunk 5]
```diff
@@ -123,18 +122,16 @@ void RobotReach::calculateAllTransformationMatricesAndCapsules(const std::vector
...
   Eigen::Matrix<double, 6, Eigen::Dynamic> jacobian1 = getJacobian(capsule, robot_capsules_for_velocity_[capsule].p1_);
   Eigen::Vector<double, 6> result1 = jacobian1 * velocity.segment(0, capsule + 1);
   RobotReach::SE3Vel vel1(result1.segment(0, 3), result1.segment(3, 3));
   // Point 2: v_2 = v_1 + \omega_1 \times (p_2 - p_1), \omega_2 = \omega_1
-  RobotReach::SE3Vel vel2(
-    vel1.first + vel1.second.cross(pointTo3dVector(
-      robot_capsules_for_velocity_[capsule].p2_ - robot_capsules_for_velocity_[capsule].p1_
-    )),
-    vel1.second);
+  RobotReach::SE3Vel vel2(vel1.first + vel1.second.cross(pointTo3dVector(robot_capsules_for_velocity_[capsule].p2_ -
```
Comment (JakobThumm) [inline safety_shield/src/robot_reach.cc]: Any reason why you reformatted this?
I liked the previous version.

**Review by JakobThumm — APPROVED** [review]
Last minor suggestions. You can merge this after this.
